client/http/client.go

```diff
@@ -78,6 +78,29 @@ func (c *Client) CreateLocalSource(ctx context.Context, datasetName string, para
 	return &source, nil
 }
 
+func (c *Client) ListSourcesByDataset(ctx context.Context, datasetName string) ([]model.Source, error) {
```

Comment:
For calls that are 1:1 with the HTTP API, would it make sense to use the autogenerated client? That should reduce the code in this PR and make it easier to keep the client behaviour consistent.

Reply:
I hear you. The goal of the handrolled client is parity between library implementation and HTTP implementation. Let's discuss in next sync whether this is valuable long term.